docs/guide/docs/developers/tutorials.md

```diff
@@ -67,10 +82,65 @@ window.botpressWebChat.init({
   showConversationsButton: true, // Whether or not to show the conversations button
   showUserName: false, // Whether or not to show the user's name
   showUserAvatar: false, // Whether or not to show the user's avatar
-  enableTranscriptDownload: false // Whether or not to show the transcript download button
+  enableTranscriptDownload: false, // Whether or not to show the transcript download button
+  externalAuthToken: 'my jwt token', // Defines a token that is sent with each messages to Botpress
+  userId: null // Allows you to override the default user id. Make sure it is not possible to guess it!
 })
 ```
 
+#### Configuring the Webchat during a conversation
+
+The method `window.botpressWebChat.configure` allows you to change the configuration of the chat during a conversation without having to reload the page
+
+## Connecting your bot with your existing backend
+
+Botpress makes it easy to communicate data securely between your backend and your bot using JWT. Store any data you'd like in the token and pass it to the webchat using the `externalAuthToken` configuration option.
+
+If the token is valid, the content will be available under `event.credentials` when processing the event. If there is no token or if it is invalid, `credentials` will stay undefined.
+
+How to enable:
+
+1. Configure `pro.externalAuth` in the file `botpress.config.json`
+2. Create a file named `key.pub` in the folder `data/global` containing the public key
+
+How to use:
+
+You can either define the token when the chat is initialized: `window.botpressWebChat.init({ externalAuthToken: 'userToken' })` or you can configure it while the conversation is ongoing: `window.botpressWebChat.configure({ externalAuthToken: 'userToken' })`.
+
+### Persisting the user's profile
+
+Once the user is authenticated, you may want to extract some informations out of the credentials to save them in the `user` state, like the First name, Last name, etc. All you need to do is set up a hook listening for a certain type of event, for example `update_profile`. Then, just select the required fields
+
+Example:
+
+```js
+if (event.type === 'update_profile') {
+  if (event.credentials) {
+    event.state.user = {
+      firstname: event.credentials.firstname,
+      lastname: event.credentials.lastname
+    }
+
+    // This tells the State Manager to persist the values you defined for `user`
+    event.setFlag(bp.IO.WellKnownFlags.FORCE_PERSIST_STATE, true)
+  } else {
+    console.log('Token seems invalid ')
+  }
+
+  // Since it's a custom event, we can safely skip the dialog engine
+  event.setFlag(bp.IO.WellKnownFlags.SKIP_DIALOG_ENGINE, true)
+}
+```
+
+Then send a custom event : `window.botpressWebChat.sendEvent({ type: 'update_profile' })`
```

Comment:
It does makes sense if you read the documentation from the beginning, since it's following the paragraph "Embedding the bot on your website" including all basic steps, including examples of sendEvent

Reply:
I can tell you, some wont read the all the basic steps. Maybe you can simply add a link to those basic steps ?

Reply:
It's on the same page on the  previous paragraph, so no